liota/dccs/iotcc.py

```diff
@@ -267,12 +268,18 @@ def _format_data(self, reg_metric):
             }]
         })
 
-    def set_organization_group_properties(self, reg_entity_name, reg_entity_id, reg_entity_type, entity_local_uuid,
-                                          properties):
-        log.info("Organization Group Properties defined for resource {0}".format(reg_entity_name))
+    def set_system_properties(self, reg_entity_obj, system_properties):
+        if isinstance(reg_entity_obj, RegisteredMetric):
+            reg_entity_obj.parent.sys_properties = system_properties
```

Comment:
This line even sets the metric's parent's system property to the one we get from its parent's system properties, got when you call set_organization_group_properties(). It even causes a loop, which does not make sense.

Reply:
The loop is to be on the safer side. If the set system properties is passed to metric object from the user code.

Reply:
As I explained in above reply, it does not make sense and useless to do this loop.

Reply:
As per my above comment I think it makes sense to have this logic in this method similar to set_properties method.

Reply:
cannot agree with this.

Reply:
I don't understand this thread. Let's discuss asap.